Fix cabal file to allow stack to build

ejconlon opened the pull request:
`stack build` now works, and `cabal build`. `cabal repl lib:lecture` works, as does `make repl`

ejconlon committed:
Fix cabal file to allow stack to build

ejconlon committed:
bump resolver

ejconlon committed:
Add makefile target to run repl

steven-wolfman committed:
Update README.md

steven-wolfman commented:
OK. So in future when I make a new lecture file, I'll put a module header at the top, and I'll add it to lecture.cabal.

Seems to work in my REPL. I updated README.md. Does it look good to you?

steven-wolfman committed:
Update .replit

FWIW, this should use make replit not cabal replit (but probably won't work beautifully with the "run" button anyway).

ejconlon committed:
remove exe

ejconlon committed:
comment

ejconlon commented:
I made the simplifications I mentioned in Slack: remove the exe in cabal, remove Main.hs, remove the Makefile, update docs to say `cabal repl` again. Works on replit with cabal. Works locally with stack and cabal.

ejconlon committed:
add doctest build too

ejconlon committed:
disable doctest

ejconlon commented:
I tried adding doctest but replit needs its sandbox updated if I do that, so I just left a comment.
